lib/factories/ppt/slides.js

```diff
@@ -229,42 +272,64 @@ class SlideFactory {
+        options = Object.assign(
+            {
+                // defaults
+                x: 100,
+                y: 100,
+                cx: 100,
+                cy: 100,
+            },
+            options
+        );
+
+        options.x = PptxUnitHelper.fromPixels(options.x);
+        options.y = PptxUnitHelper.fromPixels(options.y);
+        options.cx = PptxUnitHelper.fromPixels(options.cx);
+        options.cy = PptxUnitHelper.fromPixels(options.cy);
+
         let shapeColor = options.color || '00AA00';
 
+        if (options.textAlign === undefined) {
+            options.textAlign = 'center'; // for shapes, we always want text defaulted to the center
+        }
+
         if (!this.content[slideKey]['p:sld']['p:cSld'][0]['p:spTree'][0]['p:sp']) {
             this.content[slideKey]['p:sld']['p:cSld'][0]['p:spTree'][0]['p:sp'] = [];
         }
 
-        let newShapeBlock = PptFactoryHelper.createBaseShapeBlock(objectId, 'Shape', shapeX, shapeY, shapeCx, shapeCy);
+        let newShapeBlock = PptFactoryHelper.createBaseShapeBlock(objectId, 'Shape', options.x, options.y, options.cx, options.cy);
 
         newShapeBlock['p:spPr'][0]['a:prstGeom'][0]['$'].prst = type.name;
         newShapeBlock['p:spPr'][0]['a:solidFill'] = PptFactoryHelper.createColorBlock(shapeColor);
-        newShapeBlock['p:txBody'][0]['a:bodyPr'] = [{ $: { anchor: 'ctr', rtlCol: '0' } }];
-        newShapeBlock['p:txBody'][0]['a:p'] = [{ 'a:pPr': [{ $: { algn: 'ctr' } }] }];
+        newShapeBlock['p:txBody'][0]['a:bodyPr'] = [{ $: { rtlCol: '0' } }];
 
-        if (type.avLst !== undefined) {
-            newShapeBlock['p:spPr'][0]['a:prstGeom'][0]['a:avLst'] = [{ 'a:gd': [] }];
+        PptFactoryHelper.addParagraphPropertiesToBlock(newShapeBlock['p:txBody'][0]['a:p'][0], options);
+        PptFactoryHelper.addAvLstToBlock(newShapeBlock['p:spPr'][0]['a:prstGeom'][0], type.avLst);
+        PptFactoryHelper.addLinePropertiesToBlock(newShapeBlock['p:spPr'][0], options.line);
 
-            for (let prop in type.avLst) {
-                if (type.avLst.hasOwnProperty(prop)) {
-                    newShapeBlock['p:spPr'][0]['a:prstGeom'][0]['a:avLst'][0]['a:gd'].push({ $: { name: prop, fmla: `val ${type.avLst[prop]}` } });
-                }
-            }
-        }
+        let endParagraphRun = [{ $: { lang: 'en-US' } }];
 
         if (options.text) {
             newShapeBlock['p:txBody'][0]['a:p'][0]['a:r'] = [{ 'a:rPr': [{ $: { lang: 'en-US', smtClean: '0' } }], 'a:t': options.text }];
 
             if (options.textColor !== undefined) {
                 newShapeBlock['p:txBody'][0]['a:p'][0]['a:r'][0]['a:rPr'][0]['a:solidFill'] = PptFactoryHelper.createColorBlock(options.textColor);
             }
+
+            let textBodyPropertyBlock = newShapeBlock['p:txBody'][0]['a:bodyPr'][0];
+            let textRunPropertyBlock = newShapeBlock['p:txBody'][0]['a:p'][0]['a:r'][0]['a:rPr'][0];
+
+            if (options.fontFace) {
```

Comment:
Can we have `options.fontFace !== undefined` here for consistency .

Reply:
No because it may come in as a blank string, which we would still _not_ want to assign. That's why I did it as "if(options.fontFace)" - it covers both cases (undefined and blank).